multi_partner_learning.py

```diff
@@ -425,22 +422,45 @@ def prepare_aggregation_weights(self):
     def aggregate_model_weights(self):
         """Aggregate model weights from the list of models, with a weighted average"""
 
-        weights_per_layer = list(zip(*self.models_weights_list))
-        new_weights = list()
+        # Sklearn models weigths are tuples while keras model's weights are list, this differentiate keras vs sk-learn models
+        if type(self.models_weights_list[0]) is tuple:  # Check weights type for aggregation
```

Comment:
Ok I suppose this works, but it is difficult to understand and maintain. Wouldn't it be clearer/cleaner if you test on the model type? (LogisticRegression)

Reply:
For this specific instance, the model isn't initialized before (neither after), wich would be expensive for this single test.

I understand this isn't clear but I hope the comment above demystify the situation.

What do you think @bowni ?

Reply:
Ok, I get it. Let's keep it that way for the moment, but it will need some improvement at some point. I opened an issue on that to not forget it.